server/sql_app/crud.py

```diff
@@ -146,28 +150,52 @@ def delete_scenario(db: Session, scenario_id: int):
     db.commit()
     return STATUS_SUCCESS
 
-def create_parcel_stats(db: Session, parcel_wkt: schemas.ParcelStatsBase, job_id: int):
+
+def create_parcel_stats(db: Session, parcel_id: int, parcel_wkt: str, job_id: int):
     """Create a stats entry in parcel stats table."""
-    db_parcel_stats = models.ParcelStats(target_parcel_wkt=parcel_wkt, job_id=job_id)
+    db_parcel_stats = models.ParcelStats(
+        parcel_id=parcel_id, target_parcel_wkt=parcel_wkt, job_id=job_id)
     db.add(db_parcel_stats)
     db.commit()
     db.refresh(db_parcel_stats)
     return db_parcel_stats
 
+
+def create_parcel(db: Session, study_area_id: int,
+                  parcel_id: int, parcel_wkt: str):
+    """Create an entry in parcel table."""
+    db_parcel = models.Parcel(
+        study_area_id=study_area_id, parcel_id=parcel_id, wkt=parcel_wkt)
+    try:
+        db.add(db_parcel)
+        db.commit()
+        db.refresh(db_parcel)
+        return STATUS_SUCCESS
+    except exc.IntegrityError:
+        db.rollback()
+        return STATUS_FAIL
```

Comment:
This is the exception when trying to insert a row with a duplicate primary key. So this is handling a request from a user to add a parcel to a study area when the parcel already exists. No need to check whether that occurred, from the perspective of the requester. 

Unrelatedly, I now realize a bug where the parcel table really needs to use the combination of `parcel_id` and `study_area_id` when it is adding/removing entries. Because those add/remove should only affect the current study area.

Reply:
Ahhhh, good point!